### ISSUE TITLE
Adding MTR

### ISSUE THREAD

**Issue by agahkarakuzu** [neuropoly/qMRLab#262]
@ileppe @mathieuboudreau there is a user who would like to do MTR analysis using qMRLab. 

This should be quite straightforward. Does anyone have data and code to share with me?

**Comment by jcohenadad** [comment]
https://github.com/neuropoly/spinalcordtoolbox/blob/master/scripts/sct_compute_mtr.py#L84

**Comment by agahkarakuzu** [comment]
@jcohenadad thank you so much!

**Comment by mathieuboudreau** [comment]
@agahkarakuzu as for data – I see that you're cc'd on an email from Nikola to me on 06/08/2017, where he asked for my permission to use both MTsat and MTR images that were acquired on my brain for a jupyter notebook/binder to be used for grant applications and etc. 

If any of you still have that data, I give you consent to them. I know @ileppe was the one who originally acquired it for our g-ratio paper, if @nstikov doesn't still have it.

**Comment by agahkarakuzu** [comment]
@mathieuboudreau thank you for the consent, I will use that one. I am done with new GUI functions, which will facilitate creating models with lots of interdependent options such as qsm_sb. I am branching out from master to create `mt_ratio' model (hope this name is OK).

**Comment by mathieuboudreau** [comment]
@agahkarakuzu Cools thanks for taking care of this!

Yes that name is fine.

**Comment by mathieuboudreau** [comment]
Simply "mtr" would also be fine

**Comment by agahkarakuzu** [comment]
Just wanted to make it compatible with 'mt_sat', but not much similar, 'mt_rat' is awkward

**Comment by mathieuboudreau** [comment]
@agahkarakuzu Haha definitely not mt_rat lol

I wouldn't worry about compatibility with other names moreso than with the nomenclature for the field in general. MTR is used more often, but MT ratio is also common. mt_sat is written that way because the authors of that paper simply to decided to call their parameter MTsat instead of MTS or something like that, no other special reason as far as I can see.

**Comment by ravenep** [comment]
I also have a Matlab user interested in MT :rat: so this is very timely! Thanks for adding this

**Comment by agahkarakuzu** [comment]
@ravenep not at all! I will do a quick fix for optional input handling. Apparently, it fails if a mask is not provided.

**Comment by mathieuboudreau** [comment]
I thought so too. It was either @ileppe or @tanguyduval who designed the optional input functionality, so they might have an insight.

**Comment by agahkarakuzu** [comment]
Now that I've checked some of the implementations that takes optional inputs. Each one has case specific tweaks to ensure optional input independent functionality. I am branching out `mt_ratio_hotfix`  to solve this issue.
